File: genai/app/models/ingredient.py
from enum import Enum

from pydantic import BaseModel, Field


class Unit(Enum):
    PCS = "pcs"
    ML = "ml"
    G = "g"


class Ingredient(BaseModel):
    name: str = Field(description="Name of ingredient")
    amount: int = Field(description="Amount of ingredient")
    unit: Unit = Field(description="Unit of amount")

File: server/recipes/src/main/java/com/devops/entities/Ingredient.java
package com.devops.entities;

import jakarta.persistence.Embeddable;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Embeddable
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Ingredient {
    @NotNull
    private String name;
    @NotNull
    private int amount;
    @NotNull
    private Unit unit;
}
